src/plugins/gimbal/include/plugins/gimbal/gimbal_device.h

```diff
@@ -0,0 +1,177 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class GimbalDeviceImpl;
+class System;
+
+/**
+ * @brief the GimbalDevice class provides callbacks for listening to gimbal device
+ * commands. It also provides the ability to command a gimbal device, i. e. from
+ * a custom gimbal manager.
+ *
+ * Warning: This is used for low level gimbal device control. this API should
+ * only be used if you are 1) testing a gimbal manager or 2) writing a
+ * custom gimbal manager, i. e. for a ground station.
+ */
+class GimbalDevice : public PluginBase {
```

Comment:
I'm confused. Is this to test a gimbal device or to fake a gimbal device?
And I think it's wrong to have a plugin so something inheriting from `PluginBase` inside of the `gimbal` plugin.

Reply:
@julianoes It was supposed to test a gimbal device, but I'm holding off on it for now, thats why I haven't moved it to its own plugin/worked on it.